vizro-ai/src/vizro_ai/dashboard/_response_models/components.py

```diff
@@ -41,22 +41,28 @@ class ComponentPlan(BaseModel):
         """,
     )
 
-    def create(self, model, all_df_metadata) -> Union[vm.Card, vm.AgGrid, vm.Figure]:
+    def create(self, model, all_df_metadata) -> Tuple[Union[vm.Card, vm.AgGrid, vm.Figure], Optional[str]]:
         """Create the component."""
         try:
             if self.component_type == "Graph":
                 from vizro_ai import VizroAI
 
                 vizro_ai = VizroAI(model=model)
-
-                return vm.Graph(
-                    id=self.component_id,
-                    figure=vizro_ai.plot(
-                        df=all_df_metadata.get_df(self.df_name), user_input=self.component_description
+                result = vizro_ai._run_plot_tasks(
+                    df=all_df_metadata.get_df(self.df_name),
+                    user_input=self.component_description,
+                    chart_name=self.component_id,
+                    max_debug_retry=2,
```

Comment:
Is this parameter not something we need to fetch from the `.dashboard` code

Reply:
you mean "max_debug_retry"? I don't find it helpful to expose "max_debug_retry" in .dashboard(). For example, my experience in "_get_pydantic_model", too many retries doesn't seem to help llm converge to the correct answer. Instead it's more likely to cause panic and the response becomes very off-track. But also I don't have through test on this conclusion, happy to discuss further if you found it useful to expose number of retry.